src/gui/qgsmapcanvas.cpp

```diff
@@ -416,6 +416,13 @@ void QgsMapCanvas::setDestinationCrs( const QgsCoordinateReferenceSystem &crs )
   emit destinationCrsChanged();
 }
 
+void QgsMapCanvas::setTemporalController( QgsTemporalController *controller )
+{
+  mController = controller;
+
+  connect( mController, &QgsTemporalController::updateTemporalRange, this, &QgsMapCanvas::updateTemporalRange );
```

Comment:
This should be connected to the existing setTemporalRange slot instead. Add the call to autoRefreshTriggered() to the end of the existing slot.